### ISSUE TITLE
Fehlendes URL-Parsing führt zu Bugs

### ISSUE THREAD

**Issue by Skeeve** [SIWECOS/WS-TLS-Scanner#22]
Bei einem Testlauf, den @Lednerb durchgeführt hat sind mir in den Scanergebnissen Fehlermeldungen in Dieser Art aufgefallen:

```
'errorMessage' => {
    'translationStringId' => 'TLS_NOT_SUPPORTED',
    'placeholders' => {
    'host' => 'ofenshop.de/:143'
  }
},
'name' => 'IMAP_TLS',
```

Kann es sein, dass hier ein Bug vorliegt? Der `/` gehört da doch nicht hin…

**Comment by ic0ns** [comment]
Was schickt ihr denn an den TLS-Scanner?

**Comment by Skeeve** [comment]
Ich denke, exakt sowas hier:
```
http://a-p-m.de/
http://abba-ohg.de/
http://abbund.de/
http://abl-sursum.de/
http://abrechnung-service.de
http://achimsteudter.com
http://actoria.de/
http://adolf-fust.de/
http://african-adventure.de
http://agiv.de/
http://agm-gmbh.de
```

**Comment by Skeeve** [comment]
@Lednerb - Nach Telefonat mit @ic0ns: Die Übergebenen URLs müssen bereinigt sein. Die `/`am Schluß stören…

Problem vom `scanstart.sh`, oder?

**Comment by Lednerb** [comment]
Nein, kein Problem mit dem Script. Wir hatten das bereits beim TLS-Scan: Ich würde erwarten, dass der Übergebene Parameter vernünftig die URL parst und sich den Hostname raus nimmt. Der TLS-Scanner soll auch als eigenständiges Projekt arbeiten können, wie jeder andere Scanner auch.

In der BLA tritt der Fehler nicht auf, da hier explizit der Hostname geschickt wird:

https://github.com/SIWECOS/siwecos-business-layer/blob/d2b857c7918b135bfad5a4753cf6466f190b3543/app/SiwecosScan.php#L202-L203

und

https://github.com/SIWECOS/siwecos-business-layer/blob/d2b857c7918b135bfad5a4753cf6466f190b3543/app/Jobs/StartScanJob.php#L43-L44


In `$scan->url` ist dann in diesem Fall ausschließlich die MailDomain gespeichert.



----------------------------------

Meiner Meinung nach sollte der TLS vernünftiges URL-Parsing beinhalten.

Siehe auch:

1. https://github.com/SIWECOS/WS-TLS-Scanner/issues/17
2. https://github.com/SIWECOS/WS-Port-Scanner/issues/1
